shlomobot_pytest/utils.py

```diff
@@ -77,18 +90,67 @@ def calculate_total_deducted_score(
     return min(total_points_deducted, max_points_deducted)
 
 
-def import_pyfile(module_name: str) -> ModuleType:
+def import_pyfile(py_filename: str) -> ModuleType:
     """Extracts the module from a given filename"""
-    stripped_module_name = module_name.removesuffix(".py")
+    stripped_module_name = py_filename.removesuffix(".py")
 
     return import_module(stripped_module_name)
 
 
-def get_functions_from_files(file_list: list[str]) -> list[FunctionType]:
+def get_functions_from_files(
+    file_list: list[str],
+) -> Iterator[tuple[str, FunctionType]]:
     """
     Extracts all functions from the files in file_list
     Creates a generator that returns a filename, FunctionType object pair
     """
-    for filename in file_list:
-        module = import_pyfile(filename)
-        yield filename, extract_functions(module)
+    for py_filename in file_list:
+        module = import_pyfile(py_filename)
+        for function in extract_functions(module):
+            yield function.__name__, function
+
+
+def get_clean_function_lines(function: FunctionType, should_black=True) -> list[str]:
+    """Count the amount on non comment or docstring lines in a function code"""
+    function_code = inspect.getsource(function)
+
+    # reformats file to connect split lines and remove empty lines
+    if should_black:
+        function_code = format_str(function_code, mode=FileMode(line_length=99999))
+    split_code = list(filter(None, function_code.splitlines()))
+
+    clean_lines = [split_code[0]]
+    docstring_type = ""
+
+    for line in split_code[1:]:
+        # Do not count commented line or single line docstring
+        if not re.search(COMMENT_REGEX, line) and not re.search(
+            ONE_LINE_DOCSTRING_REGEX, line
+        ):
+            open_match = re.match(OPEN_DOCSTRING_REGEX, line)
+            close_match = re.match(CLOSE_DOCSTRING_REGEX, line)
+
+            # check if a multi-line docstring has started
+            # since there is only one line of code before a docstring (the def line)
+            # then when the docstring starts the length of the line list has to be 1
+            if open_match and docstring_type == "" and len(clean_lines) == 1:
+                docstring_type = open_match.group()
+
+            # Check if the previous multi-line docstring has closed
+            elif close_match and (close_match.group() == docstring_type):
+                docstring_type = ""
+
+            elif docstring_type == "":
+                clean_lines.append(line)
```

Comment:
Minor change just to make it more efficient and avoid running the regex on every line

```suggestion
            # check if a multi-line docstring has started
            # since there is only one line of code before a docstring (the def line)
            # then when the docstring starts the length of the line list has to be 1
            if  len(clean_lines) == 1 and docstring_type == "" and and re.match(OPEN_DOCSTRING_REGEX, line):
                docstring_type = open_match.group()

            # Check if the previous multi-line docstring has closed
            elif docstring_type:
                close_match = re.match(CLOSE_DOCSTRING_REGEX, line)
                if close_match and close_match.group() == docstring_type:
                    docstring_type = ""

            else:
                clean_lines.append(line)
```